dateutil/relativedelta.py

```diff
@@ -157,19 +160,34 @@ def __init__(self, dt1=None, dt2=None,
             self.microsecond = None
             self._has_time = 0
 
-            months = (dt1.year * 12 + dt1.month) - (dt2.year * 12 + dt2.month)
+            # Get year / month delta between the two
+            months = (dt1.year - dt2.year) * 12 + (dt1.month - dt2.month)
             self._set_months(months)
+
+            # Remove the year/month delta so the timedelta is just well-defined
+            # time units (seconds, days and microseconds)
             dtm = self.__radd__(dt2)
+
+            # If we've overshot our target, make an adjustment
             if dt1 < dt2:
-                while dt1 > dtm:
-                    months += 1
-                    self._set_months(months)
-                    dtm = self.__radd__(dt2)
+                compare = operator.gt
+                increment = 1
             else:
-                while dt1 < dtm:
-                    months -= 1
-                    self._set_months(months)
-                    dtm = self.__radd__(dt2)
+                compare = operator.lt
+                increment = -1
+
+            ii = 0
+            while compare(dt1, dtm):
```

Comment:
I'll note that this `while` loop is present in the first commit in the repo, so I'm not sure why it was added.